src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala

```diff
@@ -610,28 +612,39 @@ final case class GatewayApiServiceImpl(
     }
   }
 
-  def m2mTenantSeedFromApi(origin: String, externalId: String)(code: String): M2MTenantSeed =
-    M2MTenantSeed(ExternalId(origin, externalId), M2MAttributeSeed(code) :: Nil)
+  def m2mTenantSeedFromApi(origin: String, externalId: String, name: String)(code: String): M2MTenantSeed =
+    M2MTenantSeed(ExternalId(origin, externalId), M2MAttributeSeed(code) :: Nil, name)
 
   def enhanceEService(eService: CatalogManagementEService)(implicit contexts: Seq[(String, String)]): Future[EService] =
     for {
       tenant           <- tenantManagementService.getTenantById(eService.producerId)
-      selfcareId       <- tenant.selfcareId.toFuture(MissingSelfcareId)
-      producer         <- partyManagementService.getInstitution(selfcareId)
       latestDescriptor <- eService.latestAvailableDescriptor
       state            <- latestDescriptor.state.toModel.toFuture
       allAttributesIds = eService.attributes.allIds
       attributes <- attributeRegistryManagementService.getBulkAttributes(allAttributesIds)
       attributes <- eService.attributes.toModel(attributes.attributes).toFuture
+      category   <- extractCategoryIpa(tenant)
     } yield EService(
       id = eService.id,
-      producer = producer.toModel(tenant.id),
+      producer = tenant.toModel(category),
       name = eService.name,
       version = latestDescriptor.version,
       description = eService.description,
       technology = eService.technology.toModel,
       attributes = attributes,
       state = state
     )
```

Comment:
Add a space between methods/functions/expressions. I fear that your ide might be misconfigured, check if the formatter is `scalafmt`